src/sentry/api/serializers/models/release.py

```diff
@@ -197,6 +189,44 @@ def get_attrs(self, item_list, user, *args, **kwargs):
                 .values('release_id').annotate(new_groups=Sum('new_groups'))
                 .values_list('release_id', 'new_groups')
             )
+        return tags, group_counts_by_release
+
+    def __get_release_data_with_environment(self, project, item_list, environment):
+        if project:
+            release_project_envs = ReleaseProjectEnvironment.objects.filter(
+                project=project, release__in=item_list, environment=environment).select_related('release')
+        else:
+            release_project_envs = ReleaseProjectEnvironment.objects.filter(
+                release__in=item_list, environment=environment).select_related('release')
+
+        first_and_last_seen = {}
+        for release_project_env in release_project_envs:
+            first_and_last_seen[release_project_env.release.version] = {
+                'first_seen': release_project_env.first_seen,
+                'last_seen': release_project_env.last_seen,
+            }
+        issue_counts_by_release = dict(
+            release_project_envs.values('release_id').annotate(
+                new_issues_count=Sum('new_issues_count'))
+            .values_list('release_id', 'new_issues_count')
+        )
+        return first_and_last_seen, issue_counts_by_release
+
+    def get_attrs(self, item_list, user, *args, **kwargs):
+        project = kwargs.get('project')
+        environment = kwargs.get('environment')
```

Comment:
I know it wasn't introduced here, but we should circle back later (probably not in this review cycle) and make these arguments explicit rather than pulling them out of `**kwargs` like this.